tests/behat/behat-features/src/MediaReplicationContext.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * @file
+ * The behat context for replication tests.
+ */
+
+use Behat\Mink\Exception\ExpectationException;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Defines features from the contentpool context.
+ */
+class MediaReplicationContext extends RawDrupalContext {
+
+  /**
+   * Media entity that were created during test.
+   *
+   * @var int[]
+   *   Array of entity ids.
+   */
+  protected static $createdMediaEntities;
+
+  /**
+   * Media entity that were selected in article and replicated.
+   *
+   * @var int[]
+   *   Array of entity ids.
+   */
+  protected static $replicatedMediaEntities;
+
+  /**
+   * Media entity id that is selected.
+   *
+   * @var int
+   *   Currently selected media id in article.
+   */
+  protected static $currentlySelectedMedia;
+
+  /**
+   * Media entity id that was selected.
+   *
+   * @var int
+   *   Previously selected media id in article.
+   */
+  protected static $previouslySelectedMedia;
+
+  /**
+   * Media entity id tested for unused.
+   *
+   * @var int
+   *   Media entity id tested for unused.
+   */
+  protected static $unusedMedia;
+
+  /**
+   * I save media to array.
+   *
+   * @Then I save media to array with title :title
+   */
+  public function iSaveMediaToArray($title) {
+    $args = explode('/', $this->getSession()->getCurrentUrl());
+    $id = end($args);
+    self::$createdMediaEntities[$id] = $title;
+  }
+
+  /**
+   * @Then Save :element_selector value as currently selected
+   */
+  public function saveSelected($element_selector) {
+    $result = $this->getSession()
+      ->evaluateScript("jQuery(\"$element_selector\").val()");
+    if (!$result) {
+      throw new ExpectationException('Value is empty.', $this->getSession());
+    }
+
+    $parts = explode(':', $result);
+
+    self::$currentlySelectedMedia = $parts[1] ?? NULL;
+    self::$replicatedMediaEntities[] = $parts[1] ?? NULL;
+  }
+
+  /**
+   * Checks if first unused media title exists.
+   *
+   * @Then Unused media is not replicated
+   */
+  public function checkUnusedMedia() {
+    foreach (self::$createdMediaEntities as $id => $title) {
+      if (!in_array($id, self::$replicatedMediaEntities)) {
+        self::$unusedMedia = $id;
+        $found_element = $this->getSession()
+          ->evaluateScript("jQuery(\"*:contains('$title')\").length > 0");
+        if ($found_element) {
+          throw new ExpectationException('Element should not be found.', $this->getSession());
+        }
+        break;
+      }
+    }
+  }
+
+  /**
+   * @Given I click on :index of created media in :entity_browser
```

Comment:
should click on items identified by something human understandable, title, label, name or alt text.